src/module/actor/lancer-actor.ts

```diff
@@ -655,22 +659,33 @@ export class LancerActor extends Actor {
           flags.top_level_data["folder"] = unit_folder ? unit_folder.id : null;
           flags.top_level_data["token.name"] = owned_name;
           flags.top_level_data["permission"] = permission;
+          flags.top_level_data["token.disposition"] = CONST.TOKEN_DISPOSITIONS.NEUTRAL;
           // dep.writeback(); -- do this later, after setting active!
           synced_deployables.push(dep);
         },
         // Rename and rehome mechs
-        sync_mech: (mech: Mech) => {
+        sync_mech: async (mech: Mech) => {
           let flags = mech.Flags as FoundryFlagData<EntryType.MECH>;
           let portrait = mech.CloudPortrait || mech.Frame?.ImageUrl || "";
           let new_img = replace_default_resource(flags.top_level_data["img"], portrait);
+          
           flags.top_level_data["name"] = mech.Name;
           flags.top_level_data["folder"] = unit_folder ? unit_folder.id : null;
-          flags.top_level_data["token.name"] = data.callsign;
           flags.top_level_data["img"] = new_img;
-          flags.top_level_data["token.img"] = new_img;
           flags.top_level_data["permission"] = permission;
-          mech.writeback();
-          // TODO: Retrogrades
+          flags.top_level_data["token.img"] = new_img;
+          flags.top_level_data["token.name"] = data.callsign;
+          flags.top_level_data["token.disposition"] = this.data.token.disposition;
+          flags.top_level_data["token.actorLink"] = true;
+
+          await mech.writeback();
+
+          // If we've got a frame (which we should) check for setting Retrograde image
+          // Also check that we don't have a custom image, which would be on imgur. If so, preserve it.
```

Comment:
are all custom images on imgur? Is that a thing we should be assuming, even if it is currently true?

Reply:
For comp/con imports, yes. When you upload a picture to C/C it uploads it to imgur and hotlinks it. I think the only way to bypass that would be to edit the pilot JSON directly.

Reply:
Correct--we'll need to patch if they change that, but couldn't think of a good catchall beyond the reverse of ! "compcon-image-assets" which would similarly need a patch if changed